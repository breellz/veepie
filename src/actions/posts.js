import uuid from 'uuid';

//ADD_POST
export const addPost = (post) => ({
  type: 'ADD_POST',
  post
})

//DELETE_POST

export const deletePost = (id) => ({
  type: 'DELETE_POST',
  id
})

//EDIT_POST
export const editPost = (id, updates) => ({
  type: 'EDIT_POST',
  id,
  updates
}) 

// SET_POSTS

export const setPosts = (posts) =>({
  type: 'SET_POSTS',
  posts
})
