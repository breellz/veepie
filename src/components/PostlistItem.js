import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment'

  const PostlistItem = ({ id, createdAt, title, tags }) => (
  <div key={id} className="list-body">
  <h2 className="list-title">{title}</h2>
  <p>Created: {moment(createdAt).format('dddd, MMMM Do YYYY, h:mm a')}</p>
  <p>Tags: {tags.map((tag)=> <span className="tags">{tag}</span>)}</p>
 
 <div className="link-group">
 <Link className="tag-links" to={`/edit/${id}`}>Edit Post</Link>
  <Link className="tag-links" to={`/read/${id}`}>View Post</Link>
  </div>
  </div>
);

export default PostlistItem;