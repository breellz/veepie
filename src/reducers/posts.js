// posts Reducer

const postsReducerDefaultState = [];

export default (state = postsReducerDefaultState, action) => {
  switch(action.type){
    case 'ADD_POST':
      return [
        ...state,
        action.post
      ]

      case 'DELETE_POST':
        return state.filter(({ id })=> id !== action.id)

      case 'EDIT_POST':
       return  state.map((post)=>{
          if(post.id === action.id){
            return {
              ...post,
              ...action.updates
            }
          }return state
        })

        case 'SET_POSTS':
          return action.posts
    default: 
    return state
  }
}