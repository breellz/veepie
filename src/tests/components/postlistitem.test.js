import React from 'react'
import { shallow } from "enzyme";
import  PostlistItem from '../../components/PostlistItem';
import posts from '../fixtures/posts'

test('should render post list item with posts', ()=>{
    const wrapper = shallow(<PostlistItem {...posts[0]} />)
    expect(wrapper).toMatchSnapshot()
})