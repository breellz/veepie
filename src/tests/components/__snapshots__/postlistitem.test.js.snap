// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render post list item with posts 1`] = `
<div
  className="list-body"
  key="whe12"
>
  <h2
    className="list-title"
  >
    first post
  </h2>
  <p>
    Created: 
    Thursday, January 1st 1970, 1:00 am
  </p>
  <p>
    Tags: 
  </p>
  <div
    className="link-group"
  >
    <Link
      className="tag-links"
      replace={false}
      to="/edit/whe12"
    >
      Edit Post
    </Link>
    <Link
      className="tag-links"
      replace={false}
      to="/read/whe12"
    >
      View Post
    </Link>
  </div>
</div>
`;
